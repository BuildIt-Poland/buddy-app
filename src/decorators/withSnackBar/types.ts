export interface SnackBarProps {
  showSnackbar: (message: string) => void;
}
