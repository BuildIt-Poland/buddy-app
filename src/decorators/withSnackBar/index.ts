import withSnackBar from './withSnackBar';
export * from './types';

export default withSnackBar;
