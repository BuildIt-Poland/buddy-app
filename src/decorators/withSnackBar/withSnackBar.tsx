import React, { useState } from 'react';
import MuiSnackbar from '@material-ui/core/Snackbar';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import { SnackBarProps } from './types';

const withSnackBar = (WrappedComponent: React.FC<SnackBarProps & any>) => {
  const SnackBar: React.FC = props => {
    const [snackbar, setSnackbar] = useState({
      isOpen: false,
      message: '',
    });

    const showSnackbar = (message: string) => setSnackbar({ isOpen: true, message });

    const hideSnackbar = () => setSnackbar({ ...snackbar, isOpen: false });

    const onClose = (
      event: React.SyntheticEvent | React.MouseEvent,
      reason?: string
    ) => {
      if (reason !== 'clickaway') {
        hideSnackbar();
      }
    };

    return (
      <>
        <WrappedComponent {...props} showSnackbar={showSnackbar} />
        <MuiSnackbar
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'left',
          }}
          open={snackbar.isOpen}
          autoHideDuration={3000}
          message={<span>{snackbar.message}</span>}
          onClose={hideSnackbar}
          action={[
            <IconButton
              key='close'
              aria-label='close'
              color='inherit'
              onClick={onClose}>
              <CloseIcon />
            </IconButton>,
          ]}
        />
      </>
    );
  };
  return SnackBar;
};

export default withSnackBar;
