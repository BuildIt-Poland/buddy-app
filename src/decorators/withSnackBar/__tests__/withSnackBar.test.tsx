import React from 'react';
import { create } from 'react-test-renderer';

import withSnackBar from '../withSnackBar';

jest.mock('@material-ui/core/IconButton', () => 'IconButton');
jest.mock('@material-ui/icons/Close', () => 'CloseIcon');
jest.mock('@material-ui/core/Snackbar', () => 'Snackbar');

describe('Component - SnackBar', () => {
  const TestComponent = () => <div>Test</div>;
  const TestComponentWithSnackBar = withSnackBar(TestComponent);

  test('renders correctly', () => {
    const component = create(<TestComponentWithSnackBar />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
