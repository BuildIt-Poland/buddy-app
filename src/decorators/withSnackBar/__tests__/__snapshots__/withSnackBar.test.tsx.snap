// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - SnackBar renders correctly 1`] = `
Array [
  <div>
    Test
  </div>,
  <Snackbar
    action={
      Array [
        <IconButton
          aria-label="close"
          color="inherit"
          onClick={[Function]}
        >
          <CloseIcon />
        </IconButton>,
      ]
    }
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    autoHideDuration={3000}
    message={
      <span>
        
      </span>
    }
    onClose={[Function]}
    open={false}
  />,
]
`;
